cluster-autoscaler/cloudprovider/aws/aws_manager_test.go

```diff
@@ -521,6 +521,22 @@ func TestBuildNodeFromTemplate(t *testing.T) {
 				Max: aws.Int64(8),
 			},
 		},
+		instanceRequirements: &ec2.InstanceRequirements{
+			VCpuCount: &ec2.VCpuCountRange{
+				Min: aws.Int64(4),
+				Max: aws.Int64(8),
+			},
+			MemoryMiB: &ec2.MemoryMiB{
+				Min: aws.Int64(4),
+				Max: aws.Int64(8),
+			},
+			AcceleratorTypes:         []*string{aws.String(autoscaling.AcceleratorTypeGpu)},
+			AcceleratorManufacturers: []*string{aws.String(autoscaling.AcceleratorManufacturerNvidia)},
+			AcceleratorCount: &ec2.AcceleratorCount{
+				Min: aws.Int64(4),
+				Max: aws.Int64(8),
+			},
+		},
```

Comment:
Is this actually testing anything at the moment given the overrides override every element of the requirements?

Reply:
Good point, the test didn't make much sense with my changes. Previously this test specified `instanceRequirementsOverrides` as to make sure  the code ended up calling https://github.com/kubernetes/autoscaler/pull/6245/files#diff-22984a3a02b16ff49b2a94a43b49f3aa61c856483c3612b06b69dd51e347746fL392-L396 

This will now be performed on the cache side, so what I added was just confusing for the readability of the test. Instead I have now changed the test to use `instanceRequirements` which will be populated using `instanceRequirementsOverrides` on the cache side. The test surface should thus remain the same.